spec.html

```diff
@@ -6633,6 +6633,22 @@ <h1>Function Environment Records</h1>
                 Set the [[ThisValue]] and record that it has been initialized.
               </td>
             </tr>
+            <tr>
+              <td>
+                HasThisBinding()
+              </td>
+              <td>
+                Return *true* if this Environment Record contains a binding for `this`, regardless of whether the binding has been initialized.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                HasSuperBinding()
+              </td>
+              <td>
+                Return *true* if this Environment Record has a `super` binding.
+              </td>
+            </tr>
```

Comment:
[Table 16: Abstract Methods of Environment Records](https://tc39.es/ecma262/#table-15) declares `HasThisBinding` and `HasSuperBinding` (for all environment records).

[Table 18: Additional Methods of Function Environment Records](https://tc39.es/ecma262/#table-17) should only declare methods that are additional to those in Table 16.